src/plugins/viewport.ts

```diff
@@ -49,6 +49,18 @@ export default class Viewport implements IPlugin {
     return viewport;
   }
 
+  private getDocumentHeight(body, documentElement): number {
+     let bodyClientHeight = body ? body.clientHeight : null;
+     let bodyScrollHeight = body ? body.scrollHeight : null;
+     let bodyOffsetHeight = body ? body.offsetHeight : null;
+     let documentClientHeight = documentElement ? documentElement.clientHeight : null;
+     let documentScrollHeight = documentElement ? documentElement.scrollHeight : null;
+     let documentOffsetHeight = documentElement ? documentElement.offsetHeight : null;
+     let documentHeight = Math.max(bodyClientHeight, bodyScrollHeight, bodyOffsetHeight,
```

Comment:
>Math.max [](start = 26, length = 8)

Can any of these have negative values? If some of them are negative and some are 'null', then Math.max(-5, null) = 0